core/ui/ui.go

```diff
@@ -1,17 +1,51 @@
 package ui
 
+import "errors"
+
 // Config is the UI configuration for core.Endpoint
 type Config struct {
 	// HideHostname whether to hide the hostname in the Result
 	HideHostname bool `yaml:"hide-hostname"`
 	// DontResolveFailedConditions whether to resolve failed conditions in the Result for display in the UI
-	DontResolveFailedConditions bool `yaml:"dont-resolve-failed-conditions"`
+	DontResolveFailedConditions bool   `yaml:"dont-resolve-failed-conditions"`
+	Badge                       *Badge `yaml:"badge"`
+}
+
+type Badge struct {
+	ResponseTime *Thresholds `yaml:"response-time"`
+}
+
+type Thresholds struct {
+	Thresholds []int `yaml:"thresholds"`
+}
+
+var (
+	ErrInvalidUiBadgeTimeConfig = errors.New("invalid endpoint ui configuration - invalid badgetime configuration: You need to set all responsetime settings and they have to be in an ascending value range")
```

Comment:
`ErrInvalidUiBadgeTimeConfig` should be `ErrInvalidBadgeResponseTimeConfig`, and I'd suggest being a bit more explicit in the error string:
```
invalid response time badge configuration: expected parameter 'thresholds' to have 5 ascending numerical values
```